File: src/main/scala/kim/dencka/easy/collection/TwoSum.scala
package kim.dencka.easy.collection

object TwoSum extends App {

  val array = Array(3, 2, 4)

  def twoSum(nums: Array[Int], target: Int): Array[Int] = {
    for {
      i <- nums.indices
      j <- i + 1 until nums.length
    } if (nums(i) + nums(j) == target) {
      return Array(i, j)
    }
    Array()
  }

  println(twoSum(array, 6).toList)

}

File: src/main/scala/kim/dencka/java/easy/collection/TwoSum.java
package kim.dencka.java.easy.collection;

import java.util.Arrays;

public class TwoSum {
    public static void main(String[] args) {
        int[] nums = {0, 0, 0, 0, 2, 0, 0, 0, 0, 6, 0, 0};
        System.out.println(Arrays.toString(twoSum(nums, 8)));
    }

    public static int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[i] + nums[j] == target) {
                    return new int[]{i, j};
                }
            }
        }
        return null;
    }
}
